src/AdaptiveHoughGpuKernel.cpp

```diff
@@ -17,15 +20,44 @@ namespace HelixSolver
     {
         CDEBUG(DISPLAY_BASIC, " .. AdaptiveHoughKernel initiated for subregion " << idx[0] << " " << idx[1]);
 
-        constexpr float INITIAL_X_SIZE = ACC_X_SIZE / ADAPTIVE_KERNEL_INITIAL_DIVISIONS;
-        constexpr float INITIAL_Y_SIZE = ACC_Y_SIZE / ADAPTIVE_KERNEL_INITIAL_DIVISIONS;
+        const uint16_t phiRegionIndex = idx[0]; 
+        // TODO test if we need to widen phi a bit
+        const Reg phiRegion = region(-3.1415, 3.1415, phiRegionIndex, REGIONS_IN_PHI);
+        ASSURE_THAT(phiRegionIndex < REGIONS_IN_PHI, "overrun in phi index, likely confused dimensions");
+        DEBUG("... phi Region " << phiRegion.center << " +- " << phiRegion.width);
+
+        const uint16_t etaRegionIndex = idx[1];        
+        const Reg etaRegion = region(-3, 3, etaRegionIndex, REGIONS_IN_ETA);
+        ASSURE_THAT(etaRegionIndex < REGIONS_IN_ETA, "overrun in eta index, likely confused dimensions");
+        DEBUG("... eta Region " << etaRegion.center << " +- " << etaRegion.width);
+        Wedge region( phiRegion, Reg({-20, 20}  ), etaRegion); // +- 20 mm along z        
+        RegionData regionData;
+        {
+            uint16_t count = 0;
+            for ( uint16_t i = 0; i < rs.size(); ++i ) {
+                ASSURE_THAT(count < MAX_SP_PER_REGION, "Region memory not sufficient for SPs");
+                if ( region.in_wedge_r_phi_z( rs[i], phis[i], zs[i]) ) {
+                    regionData.one_over_RA[count] = INVERSE_A/(rs[i]);
+                    regionData.phi[count] = phis[i];
+                    count++;     
+                }
+            }
+            regionData.spInRegion = count;
+        }
+        DEBUG("Count of SP per region: " << regionData.spInRegion << " all SP " << rs.size() 
+                << " fraction:" << static_cast<double>(regionData.spInRegion)/rs.size());
+
+        // TODO rework splitting into pt (likely eliminate further splitting in phi)                
+        // constexpr float INITIAL_X_SIZE = ACC_X_SIZE / ADAPTIVE_KERNEL_INITIAL_DIVISIONS;
+        // constexpr float INITIAL_Y_SIZE = ACC_Y_SIZE / ADAPTIVE_KERNEL_INITIAL_DIVISIONS;
+        constexpr float INITIAL_X_SIZE = ACC_X_SIZE;
+        constexpr float INITIAL_Y_SIZE = ACC_Y_SIZE;
 
         const double xBegin = PHI_BEGIN + INITIAL_X_SIZE * idx[0];
         const double yBegin = Q_OVER_PT_BEGIN + INITIAL_Y_SIZE * idx[1];
```

Comment:
idx[1] corresponds to REGIONS_IN_ETA. Why coordinates of the accumulator cell in q/pt are based on the index of a wedge in eta? And won't this line cause yBegin to be out of range for the accumulator range since idx[1] < 9 and INITIAL_Y_SIZE = ACC_Y_SIZE = PT_BEGIN - PT_END?